main/src/ca/uwaterloo/flix/runtime/interpreter/Channel.java

```diff
@@ -99,14 +104,27 @@ public static SelectChoice select(Channel[] channels, boolean hasDefault) {
         selectLock.lock();
 
         try {
-          // Check if any channel has an element
-          for (int index = 0; index < channels.length; index++) {
```

Comment:
Ignoring performance for the moment, would the following perhaps be sufficient?:

- Copy `channels` to `shuffledChannels`
- Call `Collections.shuffle(shuffledChannels)`
- Iterate through this new shuffle array?

This avoids the issue with having the guess the maximum number of waiting channels :)

Does it make sense?

Reply:
Perhaps we don't even need to copy the channel collection (not sure)

Reply:
I think the solution makes sense. (It is definitely much simpler.) However, I don't understand the issue with guessing the maximum number of waiting channels. I don't think it would affect correctness, only use less space in the runtime for most uses of `select` statements. (I also could remove this and use the default initialization length of 10.)

Regardless, if you still think it's a good way to go, I'm on board with implementing what you suggested. I think its simplicity makes it more desirable than my current implementation.

Reply:
I misunderstood the role of `7`. It is just the initial size of the array, not the total size of it. So disregard that :)

But yeah, I think I would prefer the straightforward solution of just shuffling the array. I think it makes it more clear what is going on here. And this is already a high complexity class, so doing things the obviously correct way is a win!